javascript/core/src/verbs/onehot.ts

```diff
@@ -22,22 +22,27 @@ export interface OnehotArgs extends InputColumnArgs {
  */
 export const onehotStep: ColumnTableStep<OnehotArgs> = (
 	input,
-	{ column, prefix },
+	{ columns, prefix },
 ) => {
-	// note that this ignores potential grouping
-	// TODO: should this only apply to string column types?
-	const distinct = input
-		.rollup({
-			distinct: op.array_agg_distinct(column),
-		})
-		.get('distinct', 0) as any[]
-
-	const args = distinct.sort().reduce((acc, cur) => {
-		acc[prefix ? `${prefix}${cur}` : cur] = escape((d: any) =>
-			d[column] === null ? null : d[column] === cur ? 1 : 0,
-		)
-		return acc
-	}, {} as Record<string, ExprObject>)
+	const args = columns.reduce((acc, column) => {
+		// note that this ignores potential grouping
+		// TODO: should this only apply to string column types?
+
+		const distinct = input
+			.rollup({
+				distinct: op.array_agg_distinct(column),
+			})
+			.get('distinct', 0) as any[]
+
+		const colArgs = distinct.sort().reduce((acc, cur) => {
+			acc[prefix ? `${prefix}${column}_${cur}` : `${column}_${cur}`] = escape(
+				(d: any) => (d[column] === null ? null : d[column] === cur ? 1 : 0),
+			)
+			return acc
+		}, {} as Record<string, ExprObject>)
+
+		return { ...acc, ...colArgs }
+	}, {})
```

Comment:
Are you getting type warnings here? Should be `{} as Record<string, ExprObject>`